### ISSUE TITLE
[composite] failed to parse field [sources]

### ISSUE THREAD

**Issue by liaoquanlei** [elastic/elasticsearch-java#534]
### Java API client version

8.6.2

### Java version

8

### Elasticsearch Version

8.6.2

### Problem description

it  can not work with this

Map<String, CompositeAggregationSource> map = new HashMap<>();
        map.put("iids", CompositeAggregationSource.of(c -> c.terms(t -> t.field("iid").order(NamedValue.of("iid",SortOrder.Desc)))));

preview it to json

![image](https://user-images.githubusercontent.com/23021073/225799991-04b6a252-5dfa-4ffb-9340-a50c748c81c4.png)

**Comment by Demantel2016** [comment]
Hey, I also have the same issue, how do you solve it? any solution for it?

**Comment by salangsangvesom** [comment]
Same here. I have the same issue too. Looking forward to the fix. 

In my case, it serialized to:
```
{
  "aggs": {
    "my_buckets": {
      "composite": {
        "sources": [
          { "date": { "date_histogram": { "field": "e", "calendar_interval": "1d",
          "order": [{"_key":"asc"}]
          } } } 
        ]
      }
    }
  }
}

```

Here is the error when sending the json request above to ElasticSearch:
```
{
  "error": {
    "root_cause": [
      {
        "type": "x_content_parse_exception",
        "reason": "[8:20] [date_histogram] order doesn't support values of type: START_ARRAY"
      }
    ],
    "type": "x_content_parse_exception",
    "reason": "[8:20] [composite] failed to parse field [sources]",
    "caused_by": {
      "type": "x_content_parse_exception",
      "reason": "[8:20] [date_histogram] order doesn't support values of type: START_ARRAY"
    }
  },
  "status": 400
}

```

**Comment by swallez** [comment]
API specificaition has been fixed in https://github.com/elastic/elasticsearch-specification/pull/2151 and the generated code has been updated. This will be part of version 9.10

**Comment by Demantel2016** [comment]
> API specificaition has been fixed in [elastic/elasticsearch-specification#2151](https://github.com/elastic/elasticsearch-specification/pull/2151) and the generated code has been updated. This will be part of version 9.10

what is meant by 9.10?   is that the elastic version or the client version?    when will it be released?
